<launch>
  <arg name="voxel_size" default="0.15"/>
  <arg name="generate_esdf" default="true" />
  <arg name="manager_name" default="nodelet_manager" />

  <node pkg="chomp_planner" type="chomp_planner_node" name="chomp_node"
  output="screen">
  <param name="tsdf_voxel_size" value="$(arg voxel_size)" />
  <param name="object_function_lambda" value="0.20" />
  <param name="obstacle_cost_threshold" value="2.5" />
  <param name="update_coeff" value="0.01" />
  <param name="iteration_conversion_min" value="0.001" />
  <param name="max_iteration" value="4000" />
  <param name="collision_threshold" value="1.0" />
  <param name="trajectory_dt" value="0.1" />
  <param name="chomp_initial_noise_variance" value="0.10" />
  <param name="chomp_trial_iteration" value="3" />
  <param name="chomp_trial_break" value="true" />
  <param name="chomp_start_sphere_length" value="1.0" />
  <param name="chomp_start_sphere_time" value="0.5" />
  <param name="fit_curve" value="false" />
  <param name="velocity" value="1.0" />

  <remap from="/chomp_node/esdf_map" to="/voxblox_node/esdf_map"/>
  <remap from="/chomp_node/odometry" to="/firefly/ground_truth/odometry"/>
  <remap from="/chomp_node/goal_pose" to="/firefly/command_pose"/>
  <!-- <param name="tsdf_voxels_per_side" value="16" />    -->
  <!-- <param name="voxel_carving_enabled" value="true" /> -->
  <!-- <param name="color_mode" value="color" />           -->
  <!-- <param name="max_ray_length_m" value="3.0" />       -->
  <!-- <param name="esdf_max_distance_m" value="1.0" />    -->
  </node>

</launch>
